core/transformations.h

```diff
@@ -1914,6 +1914,88 @@ void radialAverageAxis(const MultidimArray< T >& in, char axis, MultidimArray< d
 		REPORT_ERROR(ERR_ARG_INCORRECT,"Not implemented yet");
 }
 
+template<typename T>
+void radialAverageNonCubic(const MultidimArray< T >& m,
+                   Matrix1D< int >& center_of_rot,
+                   MultidimArray< T >& radial_mean,
+                   MultidimArray< int >& radial_count,
+                   const bool& rounding = false)
+{
+    Matrix1D< double > idx(3);
+
+    size_t sizemax = std::max(XSIZE(m), YSIZE(m));
+    sizemax = std::max(ZSIZE(m), sizemax);
+    double scalex = XSIZE(m)/sizemax;
+    double scaley = YSIZE(m)/sizemax;
+    double scalez = ZSIZE(m)/sizemax;
+
+    // If center_of_rot was written for 2D image
+    if (center_of_rot.size() < 3)
+        center_of_rot.resize(3);
+
+    // First determine the maximum distance that one should expect, to set the
+    // dimension of the radial average vector
+    MultidimArray< int > distances(8);
+
+    double z = scalez * (STARTINGZ(m) - ZZ(center_of_rot));
```

Comment:
I would prefer 
```
const double z0 = ...;
const double zf = ...;
...
distances 0 = floor(sqrt(x0*x0 + y0*y0 + z0*z0));
```

Reply:
I don't get this, if it is just for clarification the comments were my own annotations and they are not relevant (I have removed them)

Reply:
I mean that I would prefer to have non-mutable (`const`) variables `x0` and `xf` instead of reusing `x` (and the same for `y` and `z`). It makes the code easier to read / understand.